pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -288,6 +300,22 @@ func enrichClusterSystemData(doc *api.OpenShiftClusterDocument, systemData *api.
 	}
 }
 
+func persistIdentityUrl(cluster *api.OpenShiftCluster, identityURL []string) error {
+	if identityURL == nil {
+		return nil
+	}
+
+	if len(identityURL) != 1 {
+		return fmt.Errorf("identityURL cannot contain multiple values: %v", identityURL)
+	}
+
+	if cluster.Identity != nil && cluster.Properties.ServicePrincipalProfile == nil {
+		cluster.Identity.IdentityURL = identityURL[0]
+	}
+
+	return nil
+}
```

Comment:
I think the code will proceed without erroring if the cluster is MIWI and no IdentityURL is passed. Is that intentional?

Reply:
I think we don't want to block an update on having that header provided or not provided, just persist when provided. What if MSI-RP behavior around secret headers changes in the future? I'm open to opinions on that.

Reply:
Ensuring that we don't block updates makes sense, but what about creates? Is the IdentityURL necessary there?

Reply:
If cluster is wi/mi and no identityURL is provided by ARM that would fail an installation because the RP wouldn't have permissions to federate identities. I can think of how to represent that in this logic and adjust things.